nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -0,0 +1,121 @@
+    String outputDirectory = config.outputDirectory;
+    this.errorOutputPath = Paths.get(outputDirectory, "errors.tsv");
+    this.suggestedFixesOutputPath = Paths.get(outputDirectory, "fixes.tsv");
+    initializeOutputFiles(config);
+  }
+
+  /**
+   * Appends the string representation of the {@link SuggestedFixInfo}.
+   *
+   * @param suggestedFixInfo SuggestedFixInfo object.
+   * @param enclosing Flag to control if enclosing method and class should be included.
+   */
+  public void serializeSuggestedFixInfo(SuggestedFixInfo suggestedFixInfo, boolean enclosing) {
+    if (enclosing) {
+      suggestedFixInfo.initEnclosing();
+    }
+    appendToFile(suggestedFixInfo.tabSeparatedToString(), suggestedFixesOutputPath);
+  }
+
+  /**
+   * Appends the string representation of the {@link ErrorMessage}.
+   *
+   * @param errorInfo ErrorMessage object.
+   */
+  public void serializeErrorInfo(ErrorInfo errorInfo) {
+    errorInfo.initEnclosing();
+    appendToFile(errorInfo.tabSeparatedToString(), errorOutputPath);
+  }
+
+  /** Cleared the content of the file if exists and writes the header in the first line. */
+  private void initializeFile(Path path, String header) {
+    try {
+      Files.deleteIfExists(path);
+    } catch (IOException e) {
+      throw new RuntimeException("Could not clear file at: " + path);
+    }
+    try (OutputStream os = new FileOutputStream(path.toFile())) {
+      header += "\n";
+      os.write(header.getBytes(Charset.defaultCharset()), 0, header.length());
+      os.flush();
+    } catch (IOException e) {
+      throw new RuntimeException(
+          "Could not finish resetting File at Path: " + path + ", Exception: " + e);
+    }
+  }
+
+  /** Initializes every file which will be re-generated in the new run of NullAway. */
+  private void initializeOutputFiles(FixSerializationConfig config) {
+    try {
+      Files.createDirectories(Paths.get(config.outputDirectory));
+      if (config.suggestEnabled) {
+        initializeFile(suggestedFixesOutputPath, SuggestedFixInfo.header());
+      }
+      initializeFile(errorOutputPath, ErrorInfo.header());
+    } catch (IOException e) {
+      throw new RuntimeException("Could not finish resetting serializer: " + e);
+    }
+  }
+
+  private void appendToFile(String row, Path path) {
+    // Since there is no method available in API of either javac or errorprone to inform NullAway
+    // that the analysis is finished, we cannot open a single stream and flush it within a finalize
+    // method. Must open and close a new stream everytime we are appending a new line to a file.
+    if (row == null || row.equals("")) {
+      return;
+    }
+    row = row.replaceAll("\\R+", " ") + "\n";
+    try (OutputStream os = new FileOutputStream(path.toFile(), true)) {
+      os.write(row.getBytes(Charset.defaultCharset()), 0, row.length());
+      os.flush();
+    } catch (Exception e) {
+      System.err.println("Error happened for writing at file: " + path);
```

Comment:
Pass `e` as cause. Also, any more precise type than `Exception` we could use here? Even if it means 2 separate catch statements?